src/server/scripts/EasternKingdoms/zone_goldshire.cpp

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "MotionMaster.h"
+#include "ObjectAccessor.h"
+#include "GameEventMgr.h"
+#include "GameTime.h"
+#include "Creature.h"
+#include "CreatureGroups.h"
+
+enum COF_Paths
+{
+    STORMWIND_PATH = 80500,
+    GOLDSHIRE_PATH = 80501,
+    WOODS_PATH = 80502,
+    HOUSE_PATH = 80503,
+    LISA_PATH = 80700
+};
+
+enum COF_Waypoints
+{
+    STORMWIND_WAYPOINT = 57,
+    GOLDSHIRE_WAYPOINT = 32,
+    WOODS_WAYPOINT = 22,
+    HOUSE_WAYPOINT = 35,
+    LISA_WAYPOINT = 4
+};
+
+enum COF_Sounds
+{
+    BANSHEE_DEATH = 1171,
+    BANSHEEPREAGGRO = 1172,
+    CTHUN_YOU_WILL_DIE = 8585,
+    CTHUN_DEATH_IS_CLOSE = 8580,
+    HUMAN_FEMALE_EMOTE_CRY = 6916,
+    GHOSTDEATH = 3416
+};
+
+enum COF_Creatures
+{
+    NPC_DANA = 804,
+    NPC_CAMERON = 805,
+    NPC_JOHN = 806,
+    NPC_LISA = 807,
+    NPC_AARON = 810,
+    NPC_JOSE = 811
+};
+
+enum COF_Events
+{
+    EVENT_WP_START_GOLDSHIRE = 1,
+    EVENT_WP_START_WOODS = 2,
+    EVENT_WP_START_HOUSE = 3,
+    EVENT_WP_START_LISA = 4,
+    EVENT_PLAY_SOUNDS = 5,
+    EVENT_BEGIN_EVENT   = 6
+};
+
+class npc_cameron : public CreatureScript
+{
+public:
+    npc_cameron() : CreatureScript("npc_cameron") { }
+
+    struct npc_cameronAI : public ScriptedAI
+    {
+        npc_cameronAI(Creature* creature) : ScriptedAI(creature)
+        {
+            _started = false;
+            ///! @Riztazz
+            ///! only for testing, should be called elsewhere
```

Comment:
can this be moved somewhere else ?

Reply:
Prolly is not even needed, ill get back to this one after #20779 is merged.